### PR TITLE
SMT-LIB interface

### PR DESCRIPTION
**Pull request by ttuegel**
Changes:
1. Recognize the `smtlib` attribute.
1. Rewrite SMT-LIB translation using simple-smt.
1. Share one solver process throughout simplification.

See also: https://trello.com/c/Y1RtJRY4, https://trello.com/c/Hvp52S0f
###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`
- [ ] Style conformance: `stylish-haskell`

---

### CODE REVIEW

File: src/main/haskell/kore/src/Kore/ASTVerifier/AttributesVerifier.hs [hunk 1]
```diff
@@ -106,38 +102,13 @@ verifySymbolHookAttribute
     -> AttributesVerification atts
     -> Attributes
     -> Either (Error VerifyError) Hook
-verifySymbolHookAttribute indexedModule =
+verifySymbolHookAttribute _ =
```
Comment (virgil-serbanuta) [inline src/main/haskell/kore/src/Kore/ASTVerifier/AttributesVerifier.hs]: If the IndexedModule argument is, indeed, not needed, you could fully delete it.

File: src/main/haskell/kore/src/Kore/Attribute/Smtlib.hs [hunk 2]
```diff
@@ -0,0 +1,131 @@
+module Kore.Attribute.Smtlib
```
Comment (virgil-serbanuta) [inline src/main/haskell/kore/src/Kore/Attribute/Smtlib.hs]: Module doc?

File: src/main/haskell/kore/src/Kore/Attribute/Smtlib.hs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+import           Kore.AST.Common
+import           Kore.AST.Kore
+import           Kore.AST.MetaOrObject
+import           Kore.Attribute.Parser
+                 ( ParseAttributes (..), Parser )
+import qualified Kore.Attribute.Parser as Parser
+import qualified Kore.Builtin.Error as Builtin.Error
+import qualified Kore.Error
+
+{- | The @smtlib@ attribute for symbols.
```
Comment (virgil-serbanuta) [inline src/main/haskell/kore/src/Kore/Attribute/Smtlib.hs]: Could you describe its syntax somewhere?

File: src/main/haskell/kore/src/Kore/Builtin/Error.hs [hunk 4]
```diff
@@ -0,0 +1,26 @@
+module Kore.Builtin.Error
```
Comment (virgil-serbanuta) [inline src/main/haskell/kore/src/Kore/Builtin/Error.hs]: Could you add module documentation?

File: src/main/haskell/kore/test/Test/Kore/Builtin/Definition.hs [hunk 5]
```diff
@@ -0,0 +1,949 @@
...
+    -> [Sort Object]
+    -- ^ argument sorts
+    -> [CommonKorePattern]
+    -- ^ declaration attributes
+    -> KoreSentence
+hookedSymbolDecl
+    SymbolOrAlias { symbolOrAliasConstructor }
+    sentenceSymbolResultSort
+    sentenceSymbolSorts
+    (Attributes -> sentenceSymbolAttributes)
```
Comment (u--) [inline src/main/haskell/kore/test/Test/Kore/Builtin/Definition.hs]: Double take on this line...

**Pull request merged**